src/connection_manager.c

```diff
@@ -359,24 +359,21 @@ void qd_connection_manager_free(qd_connection_manager_t *cm)
     qd_config_listener_t *cl = DEQ_HEAD(cm->config_listeners);
     while (cl) {
         DEQ_REMOVE_HEAD(cm->config_listeners);
-        qd_server_listener_free(cl->listener);
         qd_server_config_free(&cl->configuration);
-        free(cl);
+        qd_config_listener_free(cm, cl);
```

Comment:
This is an additional bug I found. free(cl) and free(cc) were called instead of qd_config_listener_free(cm, cl) and qd_config_connector_free(cm, cc) respectively. I fixed that in this method.